inference-engine/include/details/ie_cnn_network_iterator.hpp

```diff
@@ -151,6 +160,26 @@ CNNNetworkIterator {
             }
         }
 
+        // check possibly for other disjoint univisited input subtree
+        if (nextLayersTovisit.empty()) {
+            InputsDataMap inputs;
+            network->getInputsInfo(inputs);
+            if (!inputs.empty()) {
```

Comment:
Why do you need this if?
Maybe will be easier to write just:
```
for (const auto& in : inputs) {
```

Reply:
Yes, will update.